### PR TITLE
[cxx interop] Export inout params as C++ reference

### PR DESCRIPTION
**Pull request by WANGJIEKE**
Support `inout` parameters as C++ references

### CODE REVIEW

**Comment by WANGJIEKE** [comment]
@hyp Could you please review this PR? Thanks.

**Comment by hyp** [comment]
That's a good start!

Here are some suggestions:
- You only need to test one function instead of copying all the primitive type functions in your test case file.
- It seems that this doesn't pass the values by pointers at the C layer. Do you need to update the generated C function signature as well?

**Comment by WANGJIEKE** [comment]
> It seems that this doesn't pass the values by pointers at the C layer. Do you need to update the generated C function signature as well?

I didn't realize the layer is in C (not C++).

**Comment by WANGJIEKE** [comment]
@hyp Hi, I've changed the `printFunctionSignature` function. Now it will print reference type if it's an `inout` param and it's in C++ inline function thunk mode. It will print as pointer type if it's an `inout` param and in C function prototype mode. Also remove some excessive test cases.

File: test/Interop/SwiftToCxx/functions/swift-primitive-inout-functions-cxx-bridging.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend %s -typecheck -module-name Functions -clang-header-expose-public-decls -emit-clang-header-path %t/functions.h
+// RUN: %FileCheck %s < %t/functions.h
+
+// RUN: %check-interop-cxx-header-in-clang(%t/functions.h)
+
+// CHECK:      inline void inOutInt(swift::Int & x) noexcept {
```
Comment (hyp) [inline test/Interop/SwiftToCxx/functions/swift-primitive-inout-functions-cxx-bridging.swift:7]: Can you also add a `CHECK` line to test that the generated C declaration that you're calling inside this stub has the `*` type?

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Review by hyp — APPROVED** [review]
LGTM

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Comment by shahmishal** [comment]
@swift-ci please test source compatibility

**Pull request merged**